src/common/board.rs

```diff
@@ -31,39 +29,61 @@ pub(crate) fn register_models(registry: &mut ComponentRegistry) {
     }
 }
 
-pub struct FakeBoard {
-    analogs: Vec<Rc<RefCell<dyn AnalogReader<u16, Error = anyhow::Error>>>>,
-    i2cs: HashMap<String, Arc<Mutex<FakeI2CHandle>>>,
-    pin_pwms: HashMap<i32, f64>,
-    pin_pwm_freq: HashMap<i32, u64>,
-}
+/// Represents the functionality of a general purpose compute board that contains various components such as analog readers and digital interrupts.
 pub trait Board: Status {
+    /// Set a pin to high or low
     fn set_gpio_pin_level(&mut self, pin: i32, is_high: bool) -> anyhow::Result<()>;
+
+    /// Return the current [BoardStatus] of the board
     fn get_board_status(&self) -> anyhow::Result<common::v1::BoardStatus>;
+
+    /// Get the state of a pin, high(`true`) or low(`false`)
     fn get_gpio_level(&self, pin: i32) -> anyhow::Result<bool>;
+
+    /// Get an [AnalogReader] by name
     fn get_analog_reader_by_name(
         &self,
         name: String,
     ) -> anyhow::Result<Rc<RefCell<dyn AnalogReader<u16, Error = anyhow::Error>>>>;
-    fn set_power_mode(
-        &self,
-        mode: component::board::v1::PowerMode,
-        duration: Option<Duration>,
-    ) -> anyhow::Result<()>;
+
+    /// Set the board to the indicated [PowerMode]
+    fn set_power_mode(&self, mode: PowerMode, duration: Option<Duration>) -> anyhow::Result<()>;
+
+    /// Get a wrapped [I2CHandle] by name.
     fn get_i2c_by_name(&self, name: String) -> anyhow::Result<I2cHandleType>;
+
     /// Return the amount of detected interrupt events on a pin. Should error if the
     /// pin has not been configured as an interrupt
     fn get_digital_interrupt_value(&self, _pin: i32) -> anyhow::Result<u32> {
         anyhow::bail!("this board does not support digital interrupts")
     }
+
+    /// Get the pin's given duty cycle, returns percentage as float between 0.0 and 1.0
     fn get_pwm_duty(&self, pin: i32) -> f64;
+
+    /// Set the pin to the given duty cycle , `duty_cycle_pct` is a float between 0.0 and 1.0
     fn set_pwm_duty(&mut self, pin: i32, duty_cycle_pct: f64) -> anyhow::Result<()>;
+
+    /// Get the PWM frequency of the pin
     fn get_pwm_frequency(&self, pin: i32) -> anyhow::Result<u64>;
+
+    /// Set the pin to the given PWM frequency (in Hz).
+    /// When frequency is 0, the board will unregister the pin and PWM channel from
+    /// the timer and removes the PWM signal.
     fn set_pwm_frequency(&mut self, pin: i32, frequency_hz: u64) -> anyhow::Result<()>;
 }
 
+/// An alias for a thread-safe handle to a struct that implements the [Board] trait
 pub type BoardType = Arc<Mutex<dyn Board>>;
 
+/// A test implementation of a generic compute board
```

Comment:
nit: Emulate functionality of a Board for test purpose. 
I think we should maybe hide this from docs

Reply:
To clarify, hide the description or hide the `FakeBoard` itself?
For the first we can just put a regular comment, for the second we can use `#[doc(hidden)]`

Reply:
let's just hide it altogether